internal/perms/grants.go

```diff
@@ -315,9 +329,17 @@ func (g *Grant) unmarshalText(ctx context.Context, grantString string) error {
 		switch kv[0] {
 		case "id":
 			g.id = kv[1]
+			if strings.Contains(g.id, ",") {
+				return errors.New(ctx, errors.InvalidParameter, op, "ID cannot contain a comma, semicolon or equals sign")
```

Comment:
This doesn't match the actual check. Not sure if you intended to check for semicolon/equal or not here.

Reply:
Good spot, I will update this error message. The parser makes it impossible for an ID to contain a semicolon or equals sign by this point.